packages/data/src/redux-store/metadata/selectors.js

```diff
@@ -131,3 +131,18 @@ export function isResolving( state, selectorName, args ) {
 export function getCachedResolvers( state ) {
 	return state;
 }
+
+/**
+ * Whether the store has any currently resolving selectors.
+ *
+ * @param {State} state Data state.
+ *
+ * @return {boolean} True if one or more selectors are resolving, false otherwise.
+ */
+export function hasResolvingSelectors( state ) {
+	return [ ...Object.values( state ) ].some( ( selectorState ) =>
+		[ ...selectorState._map.values() ].some(
```

Comment:
1. Are the copies necessary? `[ ...Object.values( state ) ].some` -> `Object.values( state ).some`, and likewise for `selectorState`.

 2. After digging around a bit, it seems like `._map` is an internal property of package [`equivalent-key-map`](https://github.com/aduth/equivalent-key-map). Right? My follow-up questions are: 1) is this the right/only way to consume it? 2) If so, shouldn't we at least have a comment explaining how and why we are using a private interface of an external package?

Reply:
The spread operators would be needed for results of `Map.prototype.values()`, because that's an iterator. But `Object.values()` returns an array, and that doesn't need the extra `[...]`. So, one of the spreads is justified, the other is not.

The `.map.values()` call indeed depends too much on the implementation details, and it seems to me it works only when the selector has exactly one argument? Because then the value is not in `resolution[ 1 ]`, but deeper in the tree. But the only remedy is to contribute a new `values()` method to `equivalent-key-map` itself.

Reply:
Note that the standard `Map` doesn't support `.some`. Only `.keys`, `.values`, and `.entries`, and unlike the similar `Object.*` methods they return iterators. Only after converting to array with `[...]` or `Array.from` you can call `.every` or `.some`.

What we'd like to do is to add `.keys`, `.values` and `.entries` to EKM. And I also suspect that their (or, more precisely, Andrew's 🙂) `.size` implementation is wrong, because it returns `._map.size`, but the `._map` is a tree with potentially many nested leaf nodes.

Reply:
Thanks for the answers, both of you. Right, I didn't take the time to understand the data structure behind `selectorState`, and indeed `.values` in the context of a map should just be an iterator.

> As mentioned above, .some() is a more optimized usage since it doesn't require iterating through all elements, which is why I prefer it to the .forEach() that EquivalentKeyMap supports.

To be clear, I think `.some` is the best idiom for the situation, and I wouldn't want to replace it with `.forEach`.

I'm fine with just removing the copy around `Object.values` and leaving the other as is. However, just a minor question: personally and intuitively, do you make a difference between `[ ...xs ]` and `Array.from( xs )`? For me, I tend to _infer_ that the _purpose_ of the former is to copy, while that of the latter is to cast (from iterator to array). If I'm the only one with this intuition, then ignore me; if not, then maybe this piece of code becomes 0.1% clearer by using `Array.from( selectorState._map.values ).some`. 😉

Reply:
TBH spread and `Array.from` are all the same for me in the current use case (they do have some differences, where `Array.from()` is a bit more powerful and supports more use cases than iterables, but here it doesn't really matter. 

Regardless, I'm happy to make this 0.1% more readable 👍 Updated in the PR.

Reply:
I usually prefer `Array.from`, because it does the same thing, but is less obscure: it's a regular function call, not a special syntax. Spread operator is more useful when spreading multiple things together.